compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs

```diff
@@ -566,7 +546,7 @@ encodeBuiltinExpr = \case
 
 encodeExpr' :: Expr -> Encode P.Expr
 encodeExpr' = \case
-    EVar v -> expr . either P.ExprSumVarStr P.ExprSumVarInternedStr <$> encodeName' unExprVarName v
+    EVar v -> expr . P.ExprSumVarInternedStr <$> encodeNameId unExprVarName v
```

Comment:
It's nice to see all these places where the logic becomes simpler due to merging fallible encodeName' into infallible encodeNameId

Reply:
Yes! This series of PRs to remove features is very satisfactory to work on :)